src/utils/get-menu-styles-and-animation-vals/get-menu-styles-and-animation-vals.ts

```diff
@@ -0,0 +1,135 @@
+import { useTimingTransition, interpolateColor } from 'react-native-redash'
+import {
+  Easing,
+  interpolate,
+  sub,
+  multiply,
+  add,
+  Node,
+} from 'react-native-reanimated'
+import { StyleSheet, Dimensions } from 'react-native'
+import type { TextStyle, ViewStyle } from 'react-native'
+
+import { sizes, colors } from '../../styles'
+
+type MenuStyles = {
+  readonly animated: ViewStyle
+  readonly overlay: ViewStyle
+  readonly mainContents: ViewStyle
+  readonly rotatedLabel: ViewStyle
+  readonly labelAligner: ViewStyle
+  readonly text: TextStyle
+}
+
+type StyleAndAnimationVals = {
+  readonly styles: MenuStyles
+  readonly menuSlideTranslation: Node<number>
+  readonly menuScale: Node<number>
+  readonly menuBackgroundColor: Node<number>
+  readonly labelOpacity: Node<number>
+  readonly labelCounterScale: Node<number>
+}
+
+export const getMenuStylesAndAnimationVals = (
+  hideMenu: boolean,
+  hideLabel: boolean,
+  stashDirection: 'RIGHT' | 'LEFT'
+): StyleAndAnimationVals => {
+  const { 10: labelProtrusion, 9: fontSize, 7: borderRadius } = sizes
+  const outwardDirectionMultiplier = stashDirection === 'RIGHT' ? 1 : -1
+  const labelRotation = stashDirection === 'RIGHT' ? '90deg' : '-90deg'
+
+  const styles = StyleSheet.create({
+    animated: {
+      ...StyleSheet.absoluteFillObject,
+      zIndex: 10,
+    },
+    overlay: {
+      ...StyleSheet.absoluteFillObject,
+      left: stashDirection === 'RIGHT' ? labelProtrusion * -1 : 0,
+      right: stashDirection === 'LEFT' ? labelProtrusion * -1 : 0,
+      flexDirection: 'row',
+      justifyContent: stashDirection === 'RIGHT' ? 'flex-start' : 'flex-end',
+      borderRadius,
+      opacity: 0.7,
+    },
+    mainContents: {
+      ...StyleSheet.absoluteFillObject,
+      flexDirection: 'row',
+      left: stashDirection === 'RIGHT' ? labelProtrusion : 0,
+      right: stashDirection === 'LEFT' ? labelProtrusion : 0,
+    },
+    rotatedLabel: {
+      overflow: 'visible',
+      alignSelf: 'center',
+      alignItems: 'center',
+      justifyContent: 'center',
+      height: labelProtrusion,
+      width: labelProtrusion,
+      transform: [{ rotate: labelRotation }],
+    },
+    labelAligner: {
+      alignItems: 'center',
+      width: 500,
+    },
+    text: {
+      fontSize,
+    },
+  })
```

Comment:
New line here would separate these constants well. A comment explaining that we're moving from styling to animation value setting would also be useful.

Reply:
Action taken